node-template/runtime/src/lib.rs

```diff
@@ -277,7 +281,11 @@ impl_runtime_apis! {
 
 	impl runtime_api::TaggedTransactionQueue<Block> for Runtime {
 		fn validate_transaction(tx: <Block as BlockT>::Extrinsic) -> TransactionValidity {
-			Executive::validate_transaction(tx)
+			match tx.is_signed() {
```

Comment:
I'd prefer not to have this `match` statement need to be duplicated across runtime implementations (it already appears twice in this PR, i would need to be copy'n'pasted to polkadot and basically every other runtime across the Substrate/Polkadot ecosystem).

Let's try to put it in `executive` module?

Reply:
Yes I agree.

So I added a new generic to executor which is UnsignedValidator that can validate unsigned transaction.

Also while doing so I wondered about supposed meaning of traits:
* a signed extrinsic is not equivalent to an applyable with an index and a sender isn't it ?

because of that I match on (uxt.is_signed(), sender, index).

Reply:
but also thoughts about introducing new structure like CheckedUnsignedExtrinsic. I don't really know what is preferred.